### DISCUSSION TITLE
Negative numbers greater than expected due to floating point error

### DISCUSSION THREAD

**Discussion by Aariq** [levisc8/ipmr#60] [Q&A] [answered]
```
Error : Negative numbers greater than expected due to  floating point error generated building: leave_sdlg. Double check model parameteriztion.
```

Can someone explain more about what this means and why this might happen?  I'm getting this error in a "leave_sdlg" kernel where seedlings (discrete) enter a mature plant cohort that is structured by size (continuous).  It only happens with certain runs of a stochastic IPM as well, so I assume it has something to do with this kernel generating plants that are too small. I thought I was correcting for that with `evict_cor` and `evict_fun` though.

**Comment by Aariq** [comment]
I think I answered my own question.  I had accidentally forgot to back-transform a survival model, so survival was often > 1.  Not entirely sure how that relates to the error message, but fixing that seems to have gotten rid of it.

**Comment by levisc8** [comment, marked as answer]
So the error message means that some value in the sub-kernel named leave_sdlg is negative, which can't be true for an IPM to work (i.e. it will produce negative individuals when multiplied by the population trait distribution). If that sub-kernel had stochastic parameters whose distributions *could* produce negative numbers, or forgetting to back transform yielded negative values on the logit scale, then that would definitely be the cause.

**Reply by Aariq** [reply]
FWIW, it took a long time for the error message to "click" for both me and my advisor.  I think we both initially thought the floating point error WAS the error, but it's really saying "floating point errors are cool, but this is not just a floating point error".  The way it's currently worded, I think it could be interpreted as saying that there are negative numbers greater than expected and that they were produced by floating point errors.

**Comment by levisc8** [comment]
Yea, I can see how those are not the most user-friendly error messages.
Tragically, I just submitted an update to CRAN last week, so these will
need to get rolled into the next iteration of ipmr. Sorry about that!
